src/viam/sdk/common/proto_t.hpp

```diff
@@ -0,0 +1,293 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace google {
+namespace protobuf {
+
+// Forward declaration of google::protobuf::Value.
+// The class below is written so as to keep Value out of the ABI, and as such can be instantiated
+// with Value as an incomplete type.
+
+class Value;
+}  // namespace protobuf
+}  // namespace google
+
+namespace viam {
+namespace sdk {
+
+// Type erased value for storing protobuf Value types.
+// The "simple" value types are empty, bool, int, double, and string.
+// Moreover, a ProtoT can be a vector or string-map of ProtoT.
+class ProtoT {
```

Comment:
I know the existing type is called ProtoType, but in google terminology this stuff is known as Struct and Value. Could we find names that are more aligned? Like `ProtoT` is `ProtoValue` and the map typename is `ProtoStruct`?

Reply:
That makes sense to me!